src/bootstrap/test.rs

```diff
@@ -514,6 +514,44 @@ impl Step for RustdocJS {
     }
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct RustdocUi {
```

Comment:
This is a lot of code for very few tests. I was hoping we could somehow express this in the regular ui tests by inserting some flags via comments in the test?

Reply:
As would i, but (as far as i know) there's no facility in compiletest to swap out which binary is called for a specific test - only which arguments are passed. In addition, we would have to tell bootstrap to have rustdoc ready for the UI tests, even if someone is trying to run a specific non-rustdoc test.

Plus, we wanted to introduce UI tests for rustdoc at some point anyway. `>_>` Having them in a separate test directory will help us when we add more over time.

Reply:
We need to have a struct directly to bind the test folder path.